File: LogicalThinkingPatterns/LetterTrianglePattern/LetterTriangle.py
class LetterTriangle:
    def printPattern(self, n):
        for i in range(0, n):
            for ch in range(ord('A'), ord('A') + i + 1):
                print(chr(ch)+" ", end ="")
            print()
    
    def run(self):
        n = int(input("Enter the number of rows: "))
        self.printPattern(n)

letterTriangle = LetterTriangle()
letterTriangle.run()

File: LogicalThinkingPatterns/LetterTrianglePattern/LetterTriangle.ts
class LetterTriangle {
  printPattern(n: number): void {
    for (let i = 0; i < n; i++) {
      let pattern: string = "";
      for (let ch = 65; ch <= 65 + i; ch++) {
        pattern += String.fromCharCode(ch) + " ";
      }
      console.log(pattern);
    }
  }
}

const letterTriangle = new LetterTriangle();
letterTriangle.printPattern(5);
